nativelink-store/src/redis_store.rs

```diff
@@ -293,6 +293,22 @@ impl RedisStore {
         })
     }
 
+    async fn get_client(&'_ self) -> Result<&'_ Client, Error> {
+        let config = self.client_pool.last().client_config();
+        if config.mocks.is_none() {
+            self.client_pool.wait_for_connect().await.err_tip(|| {
```

Comment:
I understand why this is done and it definitely makes sense, given it was silently failing before. I'm curious about the overhead of `wait_for_connect()`. It only happens once, correct?

Reply:
This is only a nit btw, and you can resolve. I'm curious in the answer only for my own understanding. Synchronous, blocking threads of execution are always interesting in high throughput scenarios like our own. I've added a fe of these recently as well, and they also piqued my interest.

Reply:
The overhead is a set of `parking_lot::RwLock` being locked for read (one per client connection) plus a little bit of clone's and things internally in `fred` associated with that. There's a trivial improvement I can do to wait for just the individual connection, not the set, which will drop it to one lock v.s. one per connection, which I'll do.

I was going to note this is only kinda needed the first time, except that's not quite true. If there's been a disconnect later on and it's going through a reconnect, we'll see the same failure case without this, so I think we need to keep at least the first level "is this connected?" there.

Reply:
I agree